packages/yarnpkg-core/sources/structUtils.ts

```diff
@@ -639,6 +639,14 @@ export function stringifyIdent(ident: Ident) {
   }
 }
 
+export function wrapIdentIntoScope(ident: Ident, scope: string) {
+  if (ident.scope) {
+    return `@${scope}/${ident.scope}__${ident.name}`;
+  } else {
+    return `@${scope}/${ident.name}`;
```

Comment:
I think JSR does not permit package names without scopes.

Reply:
Indeed, but I prefer to treat it as an implementation details, it'd make the code more confusing imo to only support particular codepaths in certain cases.